### ISSUE TITLE
aws_route.destination_cidr_block being evaluated in plan after 2.59.0 update - even if count evaluates to 0

### ISSUE THREAD

**Issue by bassmanitram** [terraform-providers/terraform-provider-aws#13003]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

0.11.14

### Affected Resource(s)

* aws_route

### Terraform Configuration Files

```hcl
resource "aws_route" "vpc_peer" {
  count = "${var.vpc_peer_owner_vpc_cidr == "" ? 0 : data.template_file.number_az.rendered}"

  route_table_id            = "${element(aws_route_table.private.*.id, count.index)}"
  destination_cidr_block    = "${var.vpc_peer_owner_vpc_cidr}"
  vpc_peering_connection_id = "${aws_vpc_peering_connection.peering.id}"
}
```

`var.vpc_peer_owner_vpc_cidr` is set to the empty string. 

### Expected Behavior

As in 2.58.0 - the plan succeeds, the apply works

### Actual Behavior

2.59.0: plan produces this output:

Error: module.main.module.vpc.aws_route.vpc_peer: expected "destination_cidr_block" to be a valid IPv4 Value, got : invalid CIDR address:

**Comment by ewbankkit** [comment]
Maybe related to https://github.com/terraform-providers/terraform-provider-aws/pull/12890?

**Comment by DrFaust92** [comment]
i just tried to reproduce this with terraform 0.12.24 and plan worked fine.
ill try with terraform 11 as well and update

**Comment by bassmanitram** [comment]
Gee you guys are responsive - thx! 
I'm pretty certain it due to #12890 - I'm no au fait with TF internals, but should those new checks be done when the effective count is 0?
We have a plan for going to 12 (Oh how I so want to leave 11 behind) but can't at the moment.

**Comment by DrFaust92** [comment]
i just tested with tf 11.14 with route being in the same config as you presented and plan works fine for me.

sharing config:

```hcl
provider "aws" {
  region = "us-west-2"
}

variable "vpc_peer_owner_vpc_cidr" {
  default = ""
}

resource "aws_vpc" "foo" {
	cidr_block = "10.1.0.0/16"
	tags = {
		Name = "terraform-testacc-route-basic"
	}
}

resource "aws_internet_gateway" "foo" {
	vpc_id = "${aws_vpc.foo.id}"

	tags = {
		Name = "terraform-testacc-route-basic"
	}
}

resource "aws_route_table" "foo" {
	vpc_id = "${aws_vpc.foo.id}"
}

resource "aws_route" "bar" {
  count = "${var.vpc_peer_owner_vpc_cidr == "" ? 0 : 1}"

	route_table_id = "${aws_route_table.foo.id}"
	destination_cidr_block = "${var.vpc_peer_owner_vpc_cidr}"
	gateway_id = "${aws_internet_gateway.foo.id}"
}
```

**Comment by bflad** [comment]
In Terraform 0.11, conditionals are not short-circuiting when evaluation is required, so it could be the data source reference causing the schema validation to still occur in the original example configuration.

**Comment by bassmanitram** [comment]
Just what I was gonna say @bflad - or roughly: What you do @DrFaust92 isn't _quite_ like what I do - that "1" is actually a data reference because we need to add it to multiple route tables. If it helps the data item looks like this:

```
# Gets all the AZs in the region
data "aws_availability_zones" "available" {}

// Caps the number of AZs to 3
data "template_file" "az_zones" {
  template = "${ (length(data.aws_availability_zones.available.names) > 3) ? join(",", slice(data.aws_availability_zones.available.names, 0, 2)) : join(",", data.aws_availability_zones.available.names)}"
}

data "template_file" "number_az" {
  template = "${length(split(",",data.template_file.az_zones.rendered))}"
}
```

**Comment by ewbankkit** [comment]
@bassmanitram @DrFaust92 @bflad I can reproduce the issue with Terraform 0.11 using full config provided.

**Comment by lcasey-aarp** [comment]
We are seeing the same issue running this in terraform cloud across multiple tf versions. Is there a workaround ?

**Comment by ewbankkit** [comment]
This is a similar problem to https://github.com/terraform-providers/terraform-provider-aws/issues/13626 - An `Optional` field being set to `""` and the validator not handling that.

**Comment by bflad** [comment]
The empty string validation for Terraform 0.11 support has been merged and will release with version 2.68.0 of the Terraform AWS Provider, likely tomorrow. 👍
